testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -294,17 +294,17 @@
   raise "Text 'Create #{product} Administrator' not found" unless page.has_content?("Create #{product} Administrator")
 end
 
-# Admin Page steps
-Given(/^I am on the Admin page$/) do
-  steps %(
-    When I am authorized as "admin" with password "admin"
-    When I follow the left menu "Admin > Setup Wizard"
-    )
+# Menu permission check
+Given(/^I am authorized for the "([^"]*)" section$/) do |section|
```

Comment:
I don't understand the need for that refactor.

- The first doubt that came to my mind is, have we more sections with different permissions?
If not, it makes sense that step? It makes sense to have a case statement with only one option?

- The second is that I don't see the reason to wrap into another step, a one-liner step, it would have multiple steps.. then maybe, but just for one step? As it is used everywhere, It can have an increase in the time execution unnecessary.


Said that I don't like the step `I am authorized as "admin" with password "admin"` as it is, I would like to do a refactor of it, my main concern is that we encapsulate and hide the `visit Capybara.app_host` which opens a headless browser, in my humble opinion we should represent it differently in our tests.

Reply:
>     * The first doubt that came to my mind is, have we more sections with different permissions?

Yes we do: try to create a simple user with the lowest permissions set, many entries will disappear. [Here](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/suse/manager/webui/menu/MenuTree.java#L50) if you grep for "withVisibility" you can see how many entries are optionable depending on permissions (*adminRoles*).

>       If not, it makes sense that step? It makes sense to have a case statement with only one option?
The `case` statement is exactly there to add more. At the moment the refactoring involved only the `Admin` section, but it would be much better to add and test more cases.

>     * The second is that I don't see the reason to wrap into another step, a one-liner step, it would have multiple steps.. then maybe, but just for one step? As it is used everywhere, It can have an increase in the time execution unnecessary.

The wrapping is needed because of the `case` statement as above and also because checking that I can see the `Admin` menu section should be ignorant in terms of *what should be done to check*. It can be as it is right now, checking specific username and password, it could be checking the visibility of another piece of code/page, or whatever else. And this gets more sense if we add more `case` options, where expect not all the different permissions will be checked by username and password, where this user is the *super* user.

> 
> Said that I don't like the step `I am authorized as "admin" with password "admin"` as it is, I would like to do a refactor of it, my main concern is that we encapsulate and hide the `visit Capybara.app_host` which opens a headless browser, in my humble opinion we should represent it differently in our tests.

Bottom line: the refactor occurred first because

*Before*
```
Given I am on the Admin page
When I follow "SUSE Products" in the content area
```
or better, exploding it it was
```
When I am authorized as "admin" with password "admin"
When I follow the left menu "Admin > Setup Wizard"
When I follow "SUSE Products" in the content area
```
so
- permission checking
- loading the "Admin > Setup Wizard" page
- loading the "SUSE Products" page

*After*
```
Given I am authorized for the "Admin" section
When I follow the left menu "Admin > Setup Wizard > SUSE Products"
```
or better, exploding
```
When I am authorized as "admin" with password "admin"
When I follow the left menu "Admin > Setup Wizard > SUSE Products"
```
so
- permission checking (same as before)
- loading directly the specific page we want to reach, without passing by the intermediate page as it was before

To me it still makes sense as I shaped it, I see benefits only instead. We should only use the `case` statement more and more in all the other cases where it makes sense to check.

In the end I still think the cucumber test steps should replicate as much as possible the user workflow. In this case I read it as the following: if I can see the "Admin" menu section, I open the "Admin > Setup Wizard > SUSE Products" menu tree and I click on it to load the "SUSE Products" page directly.

Reply:
YAGNI... when we add more roles to test the sections, only then, make sense to have the case statement.

About the "Before" and "After" explanation, I completely agree to use only one navigation step to jump directly to the page. My only concern is this new method wrapping the other.

Reply:
Thanks for the detailed explanation. No big concerns anymore, apart from the wrapping mentioned by Oscar.